distribution/packages/src/common/scripts/preinst

```diff
@@ -9,6 +9,18 @@
 #       $1=1       : indicates an new install
 #       $1=2       : indicates an upgrade
 
+# Check for these at preinst time due to failures in postinst if they do not exist
+if [ -x "$JAVA_HOME/bin/java" ]; then
+    JAVA="$JAVA_HOME/bin/java"
+else
+    JAVA=`which java`
+fi
+
+if [ ! -x "$JAVA" ]; then
+    echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME"
```

Comment:
If this block is needed we have `could not find java; set JAVA_HOME or ensure java is in PATH` in `elasticsearch-env`. Maybe we can be consistent?